src/tls.c

```diff
@@ -6163,8 +6163,16 @@ static int TLSX_Cookie_Parse(WOLFSSL* ssl, const byte* input, word16 length,
 
     /* client_hello */
     extension = TLSX_Find(ssl->extensions, TLSX_COOKIE);
-    if (extension == NULL)
-        return HRR_COOKIE_ERROR;
+    if (extension == NULL) {
+#ifdef WOLFSSL_DTLS13
+        if (ssl->options.dtls)
+            /* TODO: Should we allow a ClientHello with a valid cookie even if
```

Comment:
I think we should because this might happen in multi-thread load balancing. A different thread replied earlier but now we got the CH2 so I don't think this should be an error.